ballerina/tests/test.bal

```diff
@@ -84,6 +85,53 @@ function testCreateObject() {
     }
 }
 
+@test:Config {
+    dependsOn: [testGetObject]
+}
+function testCreatePresignedUrlGet() returns error? {
+    log:printInfo("amazonS3Client->createPresignedUrl() GET");
+    Client amazonS3Client = check new (amazonS3Config);
+    string url = check amazonS3Client->createPresignedUrl(testBucketName, fileName, GET, 3600);
+    http:Client httpClient = check new (url);
+    http:Response httpResponse = check httpClient->get(EMPTY_STRING);
+    test:assertEquals(httpResponse.statusCode, 200, "Failed to create presigned URL");
+}
+
+@test:Config {
+    dependsOn: [testGetObject]
+}
+function testCreatePresignedUrlPut() returns error? {
+    log:printInfo("amazonS3Client->createPresignedUrl() PUT");
+    Client amazonS3Client = check new (amazonS3Config);
+    string url = check amazonS3Client->createPresignedUrl(testBucketName, fileName, PUT, 3600);
+    http:Client httpClient = check new (url);
+    http:Response httpResponse = check httpClient->put(EMPTY_STRING, content);
+    test:assertEquals(httpResponse.statusCode, 200, "Failed to create presigned URL");
+}
+
+@test:Config {
+    dependsOn: [testGetObject]
+}
+function testCreatePresignedUrlWithInvalidObjectName() returns error? {
+    log:printInfo("amazonS3Client->createPresignedUrl() with invalid object name");
+    Client amazonS3Client = check new (amazonS3Config);
+    string|error url =  amazonS3Client->createPresignedUrl(testBucketName, EMPTY_STRING, GET, 3600);
+    if (url is string) {
+        test:assertFail("Expected an error but got a URL");
+    }
```

Comment:
Assert for error (+ error message) instead?

Reply:
is this alright?

```ballerina 
   if (url is error) {
        test:assertEquals(url.message(), BUCKET_NAME_ERROR_MSG);
   } else {
        test:assertFail("Expected an error but got a URL");
   }
```

Reply:
I would do `assertTrue(url is error)` and then cast to error and do the assertion just to keep things comparatively shorter and avoid the if else blocks.